internal/proxy/directors.go

```diff
@@ -246,18 +241,26 @@ func (h DirectorSetHeadersByIP) Director(request *http.Request) error {
 
 	ip := net.ParseIP(host)
 
-	for _, ipHeaders := range h {
-		if !ipHeaders.IPNet.Contains(ip) {
-			continue
+	for _, headerName := range h.allHeaders {
+		_, ok := request.Header[headerName]
```

Comment:
for check headers by map key - header name must be in canonical form, or it will false-negative result.

for example header "test" in config and "Test" - from request (in cacnonical form with capital first letter).